app/upgrade_cudos.go

```diff
@@ -1075,13 +1079,23 @@ func withdrawGenesisContractBalances(genesisData *GenesisData, manifest *Upgrade
 	return nil
 }
 
-func convertBalance(balance sdk.Coins, cudosCfg *CudosMergeConfig) (sdk.Coins, error) {
+func convertAmount(app *App, ctx sdk.Context, genesisData *GenesisData, amount sdk.Int, cudosCfg *CudosMergeConfig) (sdk.Int, error) {
+	balance := sdk.NewCoins(sdk.NewCoin(genesisData.bondDenom, amount))
+	convertedBalance, err := convertBalance(app, ctx, balance, cudosCfg)
+	if err != nil {
+		return sdk.ZeroInt(), err
+	}
+	return convertedBalance.AmountOf(app.StakingKeeper.BondDenom(ctx)), nil
+
+}
```

Comment:
Nit pick: Performance wise, this is quite hoop (it is called in the 2nd nested for loop. But I do see that it iwas obviously the easiest solution to achive what is necessary.